fvm/src/machine/mod.rs

```diff
@@ -88,23 +86,110 @@ pub trait Machine: 'static {
     fn consume(self) -> Self::Blockstore;
 }
 
-/// Execution context supplied to the machine.
-#[derive(Clone, Debug)]
+/// Network-level settings. Except when testing locally, changing any of these likely requires a
+/// network upgrade.
+#[derive(Debug, Clone)]
+pub struct NetworkConfig {
+    /// The network version at epoch
+    pub network_version: NetworkVersion,
+
+    /// The maximum call depth.
+    ///
+    /// DEFAULT: 4096
+    pub max_call_depth: u32,
+
+    /// An override for builtin-actors. If specified, this should be the CID of a builtin-actors
+    /// "manifest".
+    ///
+    /// DEFAULT: `None`
+    pub builtin_actors_override: Option<Cid>,
+
+    /// Enable actor debugging.
+    ///
+    /// DEFAULT: `false`
+    pub actor_debugging: bool,
+
+    /// The price list.
+    ///
+    /// DEFAULT: The price-list for the current network version.
+    pub price_list: PriceList,
+}
+
+impl NetworkConfig {
+    /// Create a new network config for the given network version.
+    pub fn new(network_version: NetworkVersion) -> Self {
+        NetworkConfig {
+            network_version,
+            max_call_depth: 4096,
+            actor_debugging: false,
+            builtin_actors_override: None,
+            price_list: price_list_by_network_version(network_version),
+        }
+    }
+
+    /// Enable actor debugging. This is a consensus-critical option (affects gas usage) so it should
+    /// only be enabled for local testing or as a network-wide parameter.
+    pub fn enable_actor_debugging(&mut self, enabled: bool) -> &mut Self {
+        self.actor_debugging = enabled;
+        self
+    }
+
+    /// Override actors with the specific manifest. This is primarily useful for testing, or
+    /// networks prior to NV16 (where the actor's "manifest" isn't specified on-chain).
+    pub fn override_actors(&mut self, manifest: Cid) -> &mut Self {
+        self.builtin_actors_override = Some(manifest);
+        self
+    }
+
+    /// Create a [`MachineContext`] for a given `epoch` with the specified `initial_state`.
+    pub fn for_epoch(&self, epoch: ChainEpoch, initial_state: Cid) -> MachineContext {
+        MachineContext {
+            network: self.clone(),
+            epoch,
+            initial_state_root: initial_state,
+            base_fee: TokenAmount::zero(),
+            circ_supply: fvm_shared::TOTAL_FILECOIN.clone(),
+        }
+    }
+}
+
+/// Per-epoch machine context.
+#[derive(Clone, Debug, Deref, DerefMut)]
 pub struct MachineContext {
+    /// Network-level settings.
+    #[deref]
+    #[deref_mut]
+    pub network: NetworkConfig,
```

Comment:
It's unclear if this should go inside the context (my first attempt had it as a second machine parameter), but it's convenient to specify all "context" information in the same place.